src/libraries/System.IO.Compression/tests/ZipArchive/zip_CreateTests.cs

```diff
@@ -178,6 +178,22 @@ public static void CreateNormal_VerifyDataDescriptor()
             AssertDataDescriptor(memoryStream, false);
         }
 
+        [Fact]
+        public static void CreateNormal_With2SameEntries_ThrowException()
+        {
+            using var memoryStream = new MemoryStream();
+            // We need an non-seekable stream so the data descriptor bit is turned on when saving
+            var wrappedStream = new WrappedStream(memoryStream);
+
+            // Creation will go through the path that sets the data descriptor bit when the stream is unseekable
+            using (var archive = new ZipArchive(wrappedStream, ZipArchiveMode.Create))
+            {
+                CreateEntry(archive, "A", "xxx");
+                AssertExtensions.ThrowsContains<InvalidOperationException>(() => CreateEntry(archive, "A", "yyy"),
+                    "The entry name 'A' already exists in the archive.");
```

Comment:
@danmoseley do we still have CI legs running in different languages? I once received a suggestion to only look for the variable part of the string (in this case, just the duplicate entry name), not the full exception message.

`AssertExtensions.ThrowsContains` is defined like this:

https://github.com/dotnet/runtime/blob/750965da8d5e5099be46a36480a079e7c7643d44/src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs#L23-L27

@pedrobsaila I think you just want to search for the entry name, but I suggest the added entries have a larger unique name. 'A' might be too broad. Something like:

```suggestion
                string entryName = "duplicate.txt";
                CreateEntry(archive, entryName, "xxx");
                AssertExtensions.ThrowsContains<InvalidOperationException>(() => CreateEntry(archive, entryName, "yyy"), entryName);
```